org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/TaintIssuesUpdateJob.java

```diff
@@ -0,0 +1,89 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.core.internal.jobs;
+
+import java.util.Collection;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.sonarlint.eclipse.core.SonarLintLogger;
+import org.sonarlint.eclipse.core.internal.SonarLintCorePlugin;
+import org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade;
+import org.sonarlint.eclipse.core.internal.vcs.VcsService;
+import org.sonarlint.eclipse.core.resource.ISonarLintFile;
+import org.sonarlint.eclipse.core.resource.ISonarLintIssuable;
+import org.sonarlint.eclipse.core.resource.ISonarLintProject;
+import org.sonarsource.sonarlint.core.serverconnection.DownloadException;
+import org.sonarsource.sonarlint.core.serverconnection.ProjectBinding;
+
+public class TaintIssuesUpdateJob extends Job {
+  private final ProjectBinding projectBinding;
+  private final Collection<ISonarLintIssuable> issuables;
+  private final ISonarLintProject project;
+  private final ConnectedEngineFacade engineFacade;
+
+  public TaintIssuesUpdateJob(ConnectedEngineFacade engineFacade,
+    ISonarLintProject project,
+    Collection<ISonarLintIssuable> issuables, ProjectBinding projectBinding) {
+    super("Fetch server taint issues for " + project.getName());
+    this.engineFacade = engineFacade;
+    this.projectBinding = projectBinding;
+    setPriority(DECORATE);
+    this.project = project;
+    this.issuables = issuables;
+  }
+
+  @Override
+  protected IStatus run(IProgressMonitor monitor) {
+    try {
+      for (var issuable : issuables) {
+        if (monitor.isCanceled()) {
+          return Status.CANCEL_STATUS;
+        }
+        if (issuable instanceof ISonarLintFile) {
+          var file = ((ISonarLintFile) issuable);
+          String branchName = VcsService.getServerBranch(project);
+          fetchServerTaintIssues(engineFacade, projectBinding, branchName, file, monitor);
+          SonarLintMarkerUpdater.refreshMarkersForTaint(file, branchName, engineFacade);
+        }
+      }
+      return Status.OK_STATUS;
+    } catch (Throwable t) {
+      // note: without catching Throwable, any exceptions raised in the thread will not be visible
+      SonarLintLogger.get().error("Error while fetching server taint issues", t);
+      return new Status(IStatus.ERROR, SonarLintCorePlugin.PLUGIN_ID, t.getMessage());
+    }
+  }
+
+  public static void fetchServerTaintIssues(ConnectedEngineFacade engineFacade,
+    ProjectBinding projectBinding,
+    String branchName,
+    ISonarLintFile file, IProgressMonitor monitor) {
+    var filePath = file.getProjectRelativePath();
+
+    try {
+      engineFacade.downloadAllServerTaintIssuesForFile(projectBinding, branchName, filePath, monitor);
```

Comment:
I think for VSCode we call this and also `syncServerTaintIssues` on file open, and maybe we should not

Reply:
I don't think you should. The goal is to stop fetching stuff on file open. Starting from SQ 9.6+, updates should come with SSE. `downloadAllServerTaintIssuesForFile` is kept for older SQ versions (and SC)